pkg/operator/oauthapiencryptioncontroller/oauthapiencryptioncontroller.go

```diff
@@ -0,0 +1,128 @@
+package oauthapiencryptioncontroller
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/operatorclient"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/encryption/encryptionconfig"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	operatorv1helpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+const (
+	encryptionConfigManagedBy      = "encryption.apiserver.operator.openshift.io/managed-by"
+	encryptionConfigManagedByValue = `WARNING: DO NOT REMOVE.
+Removing this annotation will render you cluster inaccessible.`
+)
+
+type oauthAPIServerController struct {
+	targetNamespace               string
+	oauthAPIServerTargetNamespace string
+
+	secretLister corev1listers.SecretNamespaceLister
+	secretClient corev1client.SecretsGetter
+
+	synchronizationEnabled bool
+	resourceSyncer         resourcesynccontroller.ResourceSyncer
+}
+
+// New creates OAuthAPIServerController that will manage encryption-config-openshift-oauth-apiserver as described in https://github.com/openshift/enhancements/blob/master/enhancements/etcd/etcd-encryption-for-separate-oauth-apis.md
+func New(
```

Comment:
do we want to add a `BZ` so that we don't forget to remove that code in `4.6` ?

Reply:
yes please

Reply:
done https://bugzilla.redhat.com/show_bug.cgi?id=1817629